### ISSUE TITLE
Docker (mail) environment settings prevent GUI configuration

### ISSUE THREAD

**Issue by aliebrich** [nextcloud/docker#1312]
Hi!

As a long-term (and very satisfied) Nextcloud user i'm working on my first installation via docker / docker-compose. Unfortunately it is way more troublesome than expected, one of my issues is the interaction between environment variables and NC GUI settings.
Let's look at this with the examples of mail settings.

As [Auto configuration via environment variables](https://github.com/nextcloud/docker#auto-configuration-via-environment-variables) states, there are several `MAIL_` or `SMTP_` variables which can be set.
These environment variables are picked up through `config/smtp.config.php`, which itself ends up in the docker image through `Dockerfile` & `entrypoint.sh`. So far so good, the environment configuration ends up in the running Nextcloud instance.

But problems arise as soon as you start changing these config parameters via the GUI. Nextcloud/GUI writes back to `config.php`. But as stated in [Multiple config.php file](https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/config_sample_php_parameters.html#multiple-config-php-file) (sic):
> These custom files are not overwritten by Nextcloud, and the values in these files take precedence over config.php.

That means (and truly is the case) that whatever you configure in your browser will never end up in your run configuration. And even worse, changes in the GUI will sometimes just disappear, as they are again read from another config file. Very confusing indeed. ;-)

I expect that these problems also arise for other files vs. env. variables, as soon as you try to configure it via GUI.

Am i missing something here or is this really a design problem with cooperation of NC/server and NC/docker?

Best greetings,
Andi

**Comment by FaMux** [comment]
You're right, setting MAIL_ env (and probanly others) is breaking the configuration via web interface. Nevertheless, setting env are great to auto deploy nextcloud docker intances.

**Comment by joshtrichards** [comment]
> That means (and truly is the case) that whatever you configure in your browser will never end up in your run configuration. And even worse, changes in the GUI will sometimes just disappear, as they are again read from another config file. Very confusing indeed. ;-)
>
> I expect that these problems also arise for other files vs. env. variables, as soon as you try to configure it via GUI.

This only applies to the MAIL/SMTP variables. I can't think of any other ones that have web UI counterparts. If you expect to manage external email via the web UI then don't do it in Docker.

> Am i missing something here or is this really a design problem with cooperation of NC/server and NC/docker?

Sort of. It's a mix of things. There are other issues covering this topic.

I will add a note to the SMTP section now, however, clarifying that the Docker values take priority over the Web UI settings.
